src/app/ui/timeline.cpp

```diff
@@ -1061,17 +1073,23 @@ bool Timeline::onProcessMessage(Message* msg)
 
         dx += static_cast<MouseMessage*>(msg)->wheelDelta().x;
 
-        if (msg->ctrlPressed())
-          dx = dz * frameBoxWidth();
-        else
-          dy = dz * layerBoxHeight();
-
-        if (msg->shiftPressed()) {
-          dx *= 3;
-          dy *= 3;
+        if (msg->altPressed()) {
+          double next_zoom = m_zoom + (dz < 0 ? -1 : 1);
+          setZoomAndUpdate(next_zoom);
```

Comment:
Wrap this with

    if (dz) {
      double next_zoom = m_zoom + (dz < 0 ? -1 : 1);
      setZoomAndUpdate(next_zoom);
    }

On OS X `dz` can be 0 (mouse wheel has an acceleration threshold, I think `wheelDelta()` point should be a `gfx::Point<double>` in a near future). Anyway without this `if (dz)` the zoom makes some jumps between -1 and 1.

---

Also I think the `dz` sign is inverted (it should be `m_zoom + (dz < 0 ? 1: -1)`), compare this behavior zooming with the mouse wheel on the sprite editor.